### DISCUSSION TITLE
Deploying the Admin UI as a separate app

### DISCUSSION THREAD

**Discussion by skid** [vendurehq/vendure#1064] [Q&A]
Are there any examples of a setup where the admin UI is deployed as a standalone application?

**Comment by michaelbromley** [comment]
Not that I know of, but if you are [compiling a custom UI using ui-devkit[(https://www.vendure.io/docs/plugins/extending-the-admin-ui/#compiling-as-a-deployment-step), you should be able to take the generated directory and serve it using whatever means you like, making sure to include a rewrite rule so that 404s get rewritten to index.html so that SPA routing works correctly.

**Comment by obaviet** [comment]
Admin UI Vendure is really very old compared to its technology platform. I wish the development team could refresh it.

**Reply by neilcamilleri** [reply]
And drop Angular for React or Vue while they are at it.

**Reply by michaelbromley** [reply]
What aspect of the admin ui is "really old"? It is using the latest versions of the various dependencies and is under active development.

Re switching to another framework, this is not likely as recreating all the functionality would probably take me more than a year on top of learning a new framework. But I did make quite some effort to allow other frameworks to interoperate.

**Reply by obaviet** [reply]
> What aspect of the admin ui is "really old"? It is using the latest versions of the various dependencies and is under active development.
> 
> Re switching to another framework, this is not likely as recreating all the functionality would probably take me more than a year on top of learning a new framework. But I did make quite some effort to allow other frameworks to interoperate.

Don't get me wrong, I'm not talking about its technology background but simply about the UI. This is a superfluous example of it. I feel this white space is superfluous. On the other hand I hope there is a notification feature
![Chụp web_9-9-2021_123312_demo vendure io](https://user-images.githubusercontent.com/4661721/132628858-b953dc6c-6673-42a0-9b73-9f42f76f0215.jpeg)

**Reply by obaviet** [reply]
You can move it to the top so that there are no more spaces and the headers are no longer empty

**Reply by obaviet** [reply]
I think this menu bar should shrink or expand. I think UI you can refer to saleor,Sylius. It's very reasonable
![Chụp web_9-9-2021_12492_demo vendure io](https://user-images.githubusercontent.com/4661721/132630260-9b28cb96-a524-4be2-81cc-008903ebc2de.jpeg)

**Reply by michaelbromley** [reply]
@obaviet ah ok got it, thanks for the feedback. Right now I use a slightly modified version of the [Clarity component library](https://clarity.design/), which I like because it is well suited for information-dense UIs and looks decent too. The main goal right now is for it to be functional, but for sure there are aspects that can be improved from an aesthetic perspective.

**Reply by obaviet** [reply]
@michaelbromley If you want to be different, you should also design nice and easy to customize fontend, because out there as far as I see all the open source have bad default fontend. Although this is against the philosophy from your point of view as the fontend is just a demo. However, it is a fact that the old or new platforms can thrive in part because of the diverse and beautiful fontend.

**Reply by obaviet** [reply]
and you can sell that fontend. Why not when we have a revenue stream

**Reply by michaelbromley** [reply]
@obaviet I totally agree. The lack of good storefront is a big barrier to entry for headless in general. I am unable to dedicate the time to create a beautiful, extensible storefront application - entire companies exist to solve that single problem and even they struggle to get it perfectly right. So my strategy is to partner with the likes of Vercel and Vue Storefront to build integrations with Vendure. In fact, in the coming weeks we should be launching the first version on the Vue Storefront integration.

**Reply by obaviet** [reply]
That is good news. I know that in the long run the platform owners will not design the fontend but focus on the backend and leave that playing field to the developers. However, our platform is just the beginning, there are not many developers yet, why don't you make it for revenue. hehe
